[erlang] add LSP support through erlang_ls

cfclavijo opened the pull request:
Basic changes to add LSP support for erlang layer using [erlang_ls](https://erlang-ls.github.io/) backend.
Supported [features](https://erlang-ls.github.io/features/) 
Example of my `dotspacemacs-configuration-layers`
```
(erlang :variables
             erlang-backend 'lsp
             erlang-root-dir "/home/carlosrendon/.otp_builds/otp_22_kred/lib/erlang"
             erlang-man-root-dir "/home/carlosrendon/.otp_builds/otp_22_kred/lib/erlang/man"
             erlang-ls-path "/home/carlosrendon/.erlang_ls/_build/default/bin/"
             erlang-fill-column 100
             company-minimum-prefix-length 1
             lsp-ui-doc-position 'bottom
             projectile-enable-caching t)
```

Thank you <3

smile13241324 commented on layers/+lang/erlang/layers.el:
```diff
@@ -0,0 +1,14 @@
+;;; layers.el --- Erlang Layer declarations File for Spacemacs
+;;
+;; Copyright (c) 2020 Carlos F. Clavijo & Contributors
```
Please use the standard copyright notice

smile13241324 reviewed (changes requested):
@cfclavijo I have just had a very brief look over your PR however I have noticed that it does not work like the other LSP related layers. Please have a look at `json` layer and try to change the layer to work in a similar fashion when it comes to setup lsp.

It is especially important to use a non bool backend variable. The variable must be initialised with `nil` which will result in choosing the backend after examining the loaded layers (includes lsp).

cfclavijo commented:
Hello, @smile13241324. I followed your comments, improved the way lsp backend is chosen, moved most of the config-related stuffs to func.el, updated docs and rebased

smile13241324 commented on layers/+lang/erlang/funcs.el:
```diff
@@ -0,0 +1,52 @@
+              :call-hooks t)
+            (setq company-minimum-prefix-length 1)
+            (setq company-idle-delay 0.3))))
+  (add-hook 'erlang-mode-hook 'company-mode))
```
This should not be directly called instead use `:append-hooks` tag of spacemacs|add-company-backends which will do the same.

smile13241324 commented on layers/+lang/erlang/funcs.el:
```diff
@@ -0,0 +1,52 @@
+              :backends company-lsp
+              :modes erlang-mode
+              :append-hooks nil
+              :call-hooks t)
```
Are you sure you require call-hooks? Normally this is not necessary.

smile13241324 reviewed (changes requested):
I have reviewed your PR and found some places you should change before this can be merged.
For details please check my comments.

smile13241324 reviewed (approved):
Look good now :), I'll merge it in a minute.

smile13241324 commented:
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop. 
Welcome to the project :).
